blockmanager.go

```diff
@@ -3043,4 +3085,188 @@ func (l *lightHeaderCtx) RelativeAncestorCtx(
 	return newLightHeaderCtx(
 		ancestorHeight, ancestor, l.store, l.headerList,
 	)
+
+}
+
+func (b *blockManager) sideLoadHeaders() {
+
+	// If sideloading is not enabled on this chain
+	// return quickly.
+	if !b.cfg.sideLoad.Enabled {
+		return
+	}
+
+	// If headers contained in the side load source are for a different chain network return
+	// immediately.
+	if b.sideLoadReader.HeadersChain() != b.cfg.ChainParams.Net {
+
+		log.Error("headers from side load file are of network %v "+
+			"and so incompatible with neutrino's current bitcoin network -- skipping side loading", b.sideLoadReader.HeadersChain())
+
+		return
+	}
+	log.Infof("Side loading headers from %v to %v", b.headerTip, b.sideLoadReader.EndHeight())
+
+	// Set headerTip to enable reader supply header, node needs
+	err := b.sideLoadReader.SetHeight(int64(b.headerTip))
+
+	if err != nil {
+		log.Errorf("error while setting height for sideload--- skipping sideloading: "+
+			"%v", err)
+
+		return
+	}
+	headerBatch := make([]headerfs.BlockHeader, 0)
+	for {
+		//Request header
+		header, headerErr := b.sideLoadReader.NextHeader()
+		// If any error occurs while fetching headers that does not indicate
+		// an end of file, return immediately.
+		if headerErr != nil && headerErr != io.EOF {
+
+			log.Errorf("error while fetching headers -- skipping sideloading %v", err)
+
+			return
+		}
+
+		var (
+			node     *headerlist.Node
+			prevNode *headerlist.Node
+		)
+
+		// Update node height if header is not nil
+		if header != nil {
+
+			// Ensure there is a previous header to compare against.
+			prevNodeEl := b.headerList.Back()
+			if prevNodeEl == nil {
+				log.Warnf("side load - Header list does not contain a previous" +
+					"element as expected -- exiting side load")
+
+				return
+			}
+
+			node = &headerlist.Node{Header: *header}
+			prevNode = prevNodeEl
+			node.Height = prevNode.Height + 1
+		}
+
+		if b.cfg.sideLoad.Verify && headerErr == nil {
+			err, passVerification := b.verifyBlockHeader(header, *prevNode)
+			if err != nil || !passVerification {
+				log.Debugf("Side Load- Did not pass verification at height %v"+
+					"-- rolling back to last verified checkpoint and skipping sideload", node.Height)
+
+				prevCheckpoint := b.findPreviousHeaderCheckpoint(
+					node.Height,
+				)
+
+				log.Infof("Rolling back to previous validated "+
+					"checkpoint at height %d/hash %s",
+					prevCheckpoint.Height,
+					prevCheckpoint.Hash)
+
+				err = b.rollBackToHeight(uint32(prevCheckpoint.Height))
+				if err != nil {
+					panic(fmt.Sprintf("Rollback failed: %s", err))
```

Comment:
never! better return an error and let the caller handle the situation.

Panic makes programs crash and devs panic :)

Reply:
That is what happens in the `handleheaders` function, just following the pattern.